tidy3d/components/data/monitor_data.py

```diff
@@ -575,6 +578,150 @@ class FluxTimeData(MonitorData):
     flux: FluxTimeDataArray
 
 
+class DiffractionData(MonitorData):
+    """Data associated with a :class:`.DiffractionMonitor`:
+    complex components of diffracted far fields.
+
+    Example
+    -------
+    >>> f = np.linspace(1e14, 2e14, 10)
+    >>> orders_x = list(range(-4, 5))
+    >>> orders_y = list(range(-6, 7))
+    >>> pol = ["s", "p"]
+    >>> wave_numbers = np.random.random((len(orders_x), len(orders_y), len(f))).tolist()
+    >>> coords = dict(f=f, orders_x=orders_x, orders_y=orders_y, polarization=pol)
+    >>> values = (1+1j) * np.random.random((len(orders_x), len(orders_y), len(pol), len(f)))
+    >>> field = DiffractionDataArray(values, coords=coords)
+    >>> monitor = DiffractionMonitor(
+    ...     center=(1,2,3), size=(np.inf,np.inf,0), freqs=f, name='diffraction',
+    ...     orders_x=orders_x, orders_y=orders_y
+    ...     )
+    >>> data = DiffractionData(
+    ...     monitor=monitor, E=field, H=field, ux=wave_numbers, uy=wave_numbers
+    ... )
+    """
+
+    monitor: DiffractionMonitor
+
+    ux: ArrayLike[float, 2] = pd.Field(
+        ...,
+        title="Normalized kx",
+        description="Local x component of wave vectors for each diffraction order and frequency, "
+        "relative to ``local_origin`` and oriented with respect to ``u_axis``, "
+        "normalized by (2*pi/lambda) where lambda is the wavelength "
+        "associated with the background medium.",
+    )
+
+    uy: ArrayLike[float, 2] = pd.Field(
```

Comment:
in general, I feel like maybe these `ux` and `uy` should just be called `kx`, `ky`. It's been slightly bugging me since we call them `kx` and `ky` in the docstrings and descriptions. Is there a reason for calling them `ux` and `uy`? Any preference or input from @momchil-flex ?